From 2e8355d68b7f60d9b494f3b66ea5ac37d394d599 Mon Sep 17 00:00:00 2001
From: zhang sanshan <b51434@freescale.com>
Date: Thu, 8 Jan 2015 09:47:12 +0800
Subject: [PATCH 4/7] MA-6096 enable selinux context to be added to ubifs
 images with mkfs.ubifs

1 change the android.mk to support selinux context
2 add option "-s" for mkfs.ubifs

Signed-off-by: zhang sanshan <b51434@freescale.com>
---
 Android.mk              | 5 +++--
 mkfs.ubifs/mkfs.ubifs.c | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/Android.mk b/Android.mk
index c452e05..e28671d 100644
--- a/Android.mk
+++ b/Android.mk
@@ -8,7 +8,7 @@ include $(CLEAR_VARS)
 
 LOCAL_CFLAGS += -O2 -Wall
 LOCAL_LDLIBS += -lz -lm 
- 
+LOCAL_STATIC_LIBRARIES := libselinux 
 LOCAL_SRC_FILES := \
        ubi-utils/src/libubi.c \
        mkfs.ubifs/mkfs.ubifs.c \
@@ -32,7 +32,8 @@ LOCAL_C_INCLUDES += \
         $(LOCAL_PATH)/ubi-utils/include \
         $(LOCAL_PATH)/ubi-utils/src \
         $(LOCAL_PATH)/lzo-2.03/include \
-        $(LOCAL_PATH)/uuid/  
+        $(LOCAL_PATH)/uuid/  \
+	external/libselinux/include 
 
 LOCAL_MODULE := mkfs_ubifs
 
diff --git a/mkfs.ubifs/mkfs.ubifs.c b/mkfs.ubifs/mkfs.ubifs.c
index 90f51ab..683d944 100644
--- a/mkfs.ubifs/mkfs.ubifs.c
+++ b/mkfs.ubifs/mkfs.ubifs.c
@@ -143,7 +143,7 @@ static struct inum_mapping **hash_table;
 /* Inode creation sequence number */
 static unsigned long long creat_sqnum;
 
-static const char *optstring = "d:r:m:o:D:h?vVe:c:g:f:P:k:x:X:j:R:l:j:U:F";
+static const char *optstring = "d:r:m:o:D:h?vVe:c:g:f:P:k:x:X:j:R:l:j:U:Fs:";
 
 static const struct option longopts[] = {
 	{"root",          1, NULL, 'r'},
-- 
1.8.0

